### ISSUE TITLE
Support entities

### ISSUE THREAD

**Issue by florianm** [ropensci/ruODK#152]
## Feature

Support all API endpoints relevant to entities. 

Test data
- [x] Use the "report a problem" and "address a problem" examples from the [entities tutorial](https://docs.getodk.org/tutorial-community-reporting/)
- [x] Add these two forms as new test forms: use form IDs as defaults, provide env var as overrides
- [x] Submit some test data
- [ ] For each of the following functions, add gate check warnings as per changelog whether the given ODKC_VERSION supports the respective function

https://docs.getodk.org/central-api-dataset-management/
- [x] https://docs.getodk.org/central-api-dataset-management/#datasets
- [x] https://docs.getodk.org/central-api-dataset-management/#dataset-metadata
- [x] https://docs.getodk.org/central-api-dataset-management/#download-dataset

https://docs.getodk.org/central-api-entity-management/  
- [ ] https://docs.getodk.org/central-api-entity-management/#entities-metadata
- [ ] https://docs.getodk.org/central-api-entity-management/#creating-an-entity
- [ ] https://docs.getodk.org/central-api-entity-management/#getting-entity-details
- [ ] https://docs.getodk.org/central-api-entity-management/#deleting-an-entity
- [ ] https://docs.getodk.org/central-api-entity-management/#updating-an-entity
- [ ] https://docs.getodk.org/central-api-entity-management/#listing-versions
- [ ] https://docs.getodk.org/central-api-entity-management/#getting-changes-between-versions
- [ ] https://docs.getodk.org/central-api-entity-management/#entity-audit-log

**Comment by roammi** [comment]
thank you @florianm for all the great work for this, I have an use case and I will be looking forward for it to try it out

**Comment by florianm** [comment]
Oh that's great to hear, @roammi! I'm carving out some time for ruODK, so things are moving forward again.

**Comment by florianm** [comment]
Entity Lists are now supported, Entities coming soon.

**Comment by roammi** [comment]
> Oh that's great to hear, @roammi! I'm carving out some time for ruODK, so things are moving forward again.
> 
> Do you feel your use case (if appropriate, simplified and de-identified) could be used in a vignette as an example to other ruODK users?

Yes I do, We will need to put it to work first, but it sounds great, I will be using the update entity function, I'll wait for the new release of ruodk to try it out, or is it a way to try [dataset_update] earlier?

**Comment by florianm** [comment]
@roammi you can try out emtitylist_update right now if you install the latest version of ruODK (see Readme) and update your ODKC_VERSION (see NEWS).

In consultation with the core team ruODK uses the term entitylist for datasets.

**Comment by roammi** [comment]
> @roammi you can try out entitylist_update right now if you install the latest version of ruODK (see Readme) and update your ODKC_VERSION (see NEWS).
> 
> In consultation with the core team ruODK uses the term entitylist for datasets.
> 
> Are you saying that you would like to use https://docs.getodk.org/central-api-entity-management/#updating-an-entity ? I will work on that one next then so you can try it out right away.

thank you @floriann I have the latest version and it worked succesfully! thank you for the hard work! I've used ruODK::entitylist_list(), ruODK::entitylist_detail() ruODK::entitylist_download()
![image](https://github.com/ropensci/ruODK/assets/11914956/c190aadb-b2a4-433b-8ab6-7cb3548481bf)

I also used  [bruno API](https://github.com/CEN-Nouvelle-Aquitaine/bruno-API-ODKCentral) and test it create, delete and update an entity succesfully so will be great to test it with ruODK. The challenge for update is the baseVersion, you have check the version and increase it by 1 each time making the update, it will be really great to do that natively in ruODK but it could be handled outside in R too

**Comment by lognaturel** [comment]
> The challenge for update is the baseVersion, you have check the version and increase it by 1 each time making the update, it will be really great to do that natively in ruODK but it could be handled outside in R too

You also have the option of using the force query parameter as described in https://docs.getodk.org/central-api-entity-management/#updating-an-entity

If your updates are bringing in values from an external source of truth, it generally makes sense to force the update. 

baseVersion exists to detect parallel updates from multiple clients which is especially likely in an offline context. If you’re running a script that is online, it should also generally be appropriate to use the force flag. If your updates are based on the prior version of an entity, you should be able to pull the entity and then update it almost immediately meaning you don’t have to worry about conflicting updates between the two actions unless your context is extremely dynamic.

**Comment by florianm** [comment]
As always, the insight is highly appreciated, @lognaturel!

I'll have to digest the above a bit. Maybe ruODK could default to use the force flag and, if any users express the need to do so, optionally allow to use the baseVersion.

**Comment by florianm** [comment]
https://github.com/ropensci/ruODK/releases/tag/v1.5.0 is out, pending support for the OData endpoints (after getting the naming right - see https://forum.getodk.org/t/ruodk-support-for-odata-entity-endpoints-naming/50037).
